src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs

```diff
@@ -41,16 +41,17 @@ internal partial class VisualStudioWorkspaceDiagnosticAnalyzerProviderService :
             // this will allow us to build once and deploy on different versions of VS SxS.
             var vsDteVersion = Version.Parse(dte.Version.Split(' ')[0]); // DTE.Version is in the format of D[D[.D[D]]][ (?+)], so we need to split out the version part and check for uninitialized Major/Minor below
             var assembly = Assembly.Load($"Microsoft.VisualStudio.ExtensionManager, Version={(vsDteVersion.Major == -1 ? 0 : vsDteVersion.Major)}.{(vsDteVersion.Minor == -1 ? 0 : vsDteVersion.Minor)}.0.0, PublicKeyToken=b03f5f7f11d50a3a");
-
-            // Get the analyzer assets for installed VSIX extensions through the VSIX extension manager.
-            var extensionManager = serviceProvider.GetService(assembly.GetType("Microsoft.VisualStudio.ExtensionManager.SVsExtensionManager"));
-            if (extensionManager == null)
+            if (assembly == null)
             {
                 // extension manager can't be null. if it is null, then VS is seriously broken.
                 // fail fast right away
                 FailFast.OnFatalException(new Exception("extension manager can't be null"));
             }
 
+            // Get the analyzer assets for installed VSIX extensions through the VSIX extension manager.
+            var extensionManager = serviceProvider.GetService(assembly.GetType("Microsoft.VisualStudio.ExtensionManager.SVsExtensionManager"));
+            Assumes.Present(extensionManager);
```

Comment:
Won't this still throw? I guess it's not obvious to me how this is helping.

Reply:
Yes, it will throw an exception. As opposed to calling `FailFast`. 😄

Reply:
well, we have other fail fast code that we crash VS if services that are from MEF doesn't exist. Workspace service, Language services, directly load from MEF services and etc.

so, not crashing here will just make some place else to crash...

if we are sure that this is shutdown case, we should make Roslyn to handle shutdown case more gracefully. like having some thing (static?) we set on shutdown and fail fast just do noop when we are in shutdown situation.

Reply:
@heejaechang It is possible that other code incorrectly assumes this service is available during shutdown, but it from what I can tell the only import site was designed to handle this case:

https://github.com/dotnet/roslyn/blob/7f809efa06ad3367dbac5971c9026433be6369b6/src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService.cs#L31

Reply:
> so, not crashing here will just make some place else to crash...

Agreed. It seems if we can't fetch this service we just should say there are no host-provided analyzers rather than just throwing exceptions at other systems.

Reply:
@sharwell: I think that AllowDefault is just saying that if a host has no host-provided analyzers, it doesn't need to implement a service only to say "I don't have any".

Reply:
I filed #35137 and #35138 as follow-up items from this.